app/modules/net_info.c

```diff
@@ -0,0 +1,133 @@
+// ***************************************************************************
+// net_info module for ESP8266 with nodeMCU
+//
+// adopted jan 2017 for commit to nodeMCU by Wolfgang Rosner
+// re-worked by Lukas Voborsky, @voborsky
+// ***************************************************************************
+
+// #define NODE_DEBUG
+ 
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+#include "mem.h"
+#include "task/task.h"
+
+#include "lwip/ip_addr.h"
+#include "espconn.h"
+#include "lwip/dns.h" 
+#include "lwip/app/ping.h"
+#include "lwip/raw.h"
+
+/* 
+ping_opt needs to be the first element of the structure. It is a workaround to pass the 
+calllback reference to the ping_received function. Pointer the ping_option structure is
+equal to the pointer to net_info_ping_t strcture. Reference to callback function is assumed
+to be just after the ping_option structure.
+*/
+typedef struct {
+  struct ping_option ping_opt; 
+  sint32_t ping_callback_ref; 
+  uint32 count;
```

Comment:
This field appears unused except in debugging.  The callback gets the `pingresp->seqno` not `count`.

Reply:
You're right. The field is not needed unless someone needed to know the counter of current ping. `seqno` increments across all performed pings. It can be reintroduced if needed but is removed for now.